2021/day11/day11.java

```diff
@@ -0,0 +1,148 @@
+            for (int k = 1; k < input.get(j).size() - 1; k++) {
+                if (input.get(j).get(k) > 9) {
+                    input.get(j).set(k, 0);
+                    increaseOctopus(input, k - 1, j);
+                    increaseOctopus(input, k + 1, j);
+                    increaseOctopus(input, k + 1, j - 1);
+                    increaseOctopus(input, k + 1, j + 1);
+                    increaseOctopus(input, k - 1, j - 1);
+                    increaseOctopus(input, k - 1, j + 1);
+                    increaseOctopus(input, k, j - 1);
+                    increaseOctopus(input, k, j + 1);
+                }
+                if (input.get(j).get(k + 1) > 9
+                        || input.get(j + 1).get(k + 1) > 9
+                        || input.get(j + 1).get(k) > 9
+                        || input.get(j + 1).get(k - 1) > 9
+                        || input.get(j - 1).get(k + 1) > 9
+                        || input.get(j - 1).get(k) > 9
+                        || input.get(j - 1).get(k - 1) > 9
+                        || input.get(j).get(k - 1) > 9) {
+                    explosionStatus = true;
+                }
+            }
+        }
+        return explosionStatus;
+    }
+
+    public static boolean isFlashReady(ArrayList<ArrayList<Integer>> input) {
+        for (int j = 1; j < input.size() - 1; j++) {
+            for (int k = 1; k < input.get(j).size() - 1; k++) {
+                if (input.get(j).get(k) > 9) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    public static int countFlashes(ArrayList<ArrayList<Integer>> input) {
+        int counter = 0;
+        for (int j = 1; j < input.size() - 1; j++) {
+            for (int k = 1; k < input.get(j).size() - 1; k++) {
+                if (input.get(j).get(k) == 0)
+                    counter++;
+            }
+        }
+        return counter;
+    }
+
+    public static boolean areAllFlashing(ArrayList<ArrayList<Integer>> input) {
+        for (int j = 1; j < input.size() - 1; j++) {
+            for (int k = 1; k < input.get(j).size() - 1; k++) {
+                if (input.get(j).get(k) != 0)
+                    return false;
+            }
+        }
+        return true;
+    }
+
+    public static void inspectOctopuses(int days) {
+        int flashCounter = 0;
+        boolean isAnyReadyToFlash = true;
+        ArrayList<ArrayList<Integer>> input = readInput("data.txt");
+        for (int i = 0; i < days; i++) {
+            initialIncrease(input);
+            isAnyReadyToFlash = true;
+            while (isAnyReadyToFlash) {
+                processFlashes(input);
+                isAnyReadyToFlash = isFlashReady(input);
+            }
+            flashCounter += countFlashes(input);
+            if (areAllFlashing(input)) {
+                System.out.println(i + 1);
+                break;
+            }
+        }
+        System.out.println(flashCounter);
```

Comment:
I am not a fan of these `println`s. I would instead move the logic of processing a step to a new method, and then wrap this method in two separate methods for each part, e.g. `countFlashes` for part 1 and `findSynchronisationStep` for part 2. They should be simple, something like

```java
static int countFlashes(int days) {
  // read input
  int flashes = 0;
  for (int i = 0; i < days; ++i) {
    flashes += performOctoStep(octoGrid);
  }
  return flashes;
}
```
```